### PR TITLE
First Commit

### PR DESCRIPTION
**Pull request by cs-sujeet-ahirrao**
implement Following Actions
- whois_history_search,
- whois_search,
- reverse_whois_search,
- reverse_dns_search,
- domain_subdoamin_discovery,
- brand_monitor,
- dns_lookup,
- ssl_certificates

### CODE REVIEW

File: whois-xml-api/info.json [hunk 1]
```diff
@@ -0,0 +1,979 @@
+{
+  "name": "whois-xml-api",
+  "label": "whoisXMLAPI",
+  "version": "1.0.0",
+  "description": "Whois XML API Provides comprehensive set of real-time and historic Whois, Domain name and DNS Data",
+  "publisher": "Fortinet",
+  "cs_approved": false,
+  "cs_compatible": false,
+  "category": "Information",
```
Comment (cs-prerna-joshi) [inline whois-xml-api/info.json]:
```suggestion
  "category": "Threat Intelligence",
```

File: whois-xml-api/info.json [hunk 2]
```diff
@@ -0,0 +1,979 @@
...
+          "description": "If present, search through activities discovered since the given date. Acceptable values: date in \"YYYY-MM-DD\" format",
+          "required": false,
+          "editable": true,
+          "visible": true,
+          "placeholder": "YYYY-MM-DD"
+        },
+        {
+          "title": "Created Date From",
+          "name": "createdDateFrom",
+          "type": "datetime",
```
Comment (cs-prerna-joshi) [inline whois-xml-api/info.json]:
```suggestion
          "type": "datetime",
```
```suggestion
          "type": "date",
```

File: whois-xml-api/info.json [hunk 3]
```diff
@@ -0,0 +1,979 @@
...
+                "tooltip": "Retrive lists all domains that have resolved to a given name server at one point in time on the Domain Name System (DNS)",
+                "description": "Retrive lists all domains that have resolved to a given name server at one point in time on the Domain Name System (DNS)"
+              }
+            ]
+          },
+          "tooltip": "Select the method from the following options using which you want to retrive reverse dns",
+          "description": "Select the method from the following options using which you want to retrive reverse dns"
+        },
+        {
+          "title": "From",
```
Comment (cs-prerna-joshi) [inline whois-xml-api/info.json:725]: can we have better name like Retrieve Results from Domain or From Domain?

File: whois-xml-api/operations.py [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+WHOISXML_WHOIS_SEARCH_ENDPOINT = 'https://www.whoisxmlapi.com/whoisserver/WhoisService'
+WHOISXML_REVERSE_WHOIS_SEARCH_ENDPOINT = 'https://reverse-whois.whoisxmlapi.com/api/v2'
+WHOISXML_REVERSE_DNS_SEARCH_ENDPOINT = {'IPv4 Address': 'https://dns-history.whoisxmlapi.com/api/v1',
+                                        'Mail Server': 'https://reverse-mx.whoisxmlapi.com/api/v1',
+                                        'Name Server': 'https://reverse-ns.whoisxmlapi.com/api/v1'}
+WHOISXML_DNS_LOOKUP_ENDPOINT = 'https://www.whoisxmlapi.com/whoisserver/DNSService?apiKey={}&domainName={}&outputFormat=JSON&type={}'
+WHOISXML_DOMAIN_SUBDOMAIN_DISCOVERY_ENDPOINT = 'https://domains-subdomains-discovery.whoisxmlapi.com/api/v1'
+WHOISXML_API_TOKEN_ENDPOINT = 'https://reverse-ip.whoisxmlapi.com/api/v1/?apiKey={}&ip=8.8.8.8'
+WHOISXML_BRAND_MONITOR_ENDPOINT = 'https://brand-alert.whoisxmlapi.com/api/v2'
+WHOISXML_SSL_CERTIFICATES_ENDPOINT = 'https://ssl-certificates.whoisxmlapi.com/api/v1'
```
Comment (cs-prerna-joshi) [inline whois-xml-api/operations.py]: can we move all constants in constants.py file.

File: whois-xml-api/operations.py [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+        self.sslVerify = config.get('verify')
+
+    def make_api_call(self, endpoint=None, method='GET', data=None, params=None):
+        try:
+            headers = {'Content-Type': 'application/json', 'Accept': 'application/json',
+                       "Authorization": "Token {}".format(self.api_key)}
+            response = requests.request(method, endpoint, params=params, headers=headers,
+                                        verify=self.sslVerify, data=json.dumps(data))
+            if response.ok:
+                logger.info('Successfully got response for url {}'.format(endpoint))
```
Comment (cs-prerna-joshi) [inline whois-xml-api/operations.py:26]: need to handle 404 response status.

File: whois-xml-api/info.json (hunk 1 above)
Comment (cs-sujeet-ahirrao) [reply whois-xml-api/info.json]: done

File: whois-xml-api/info.json (hunk 3 above)
Comment (cs-sujeet-ahirrao) [reply whois-xml-api/info.json:725]: done

File: whois-xml-api/operations.py (hunk 5 above)
Comment (cs-sujeet-ahirrao) [reply whois-xml-api/operations.py:26]: done

**Review by cs-prerna-joshi — APPROVED** [review]

**Pull request merged**